tests/tests/storage.rs

```diff
@@ -0,0 +1,203 @@
+use fuel_core::{
```

Comment:
@Dentosal I remember you mentioned that you want to add tests for the balances part of the API as well=)

Reply:
True. And also I want to refactor storage tests. Taking this back to draft.

Reply:
Added tests in df92693. Refactored existing ones in 49b3f7c.